Fix NPE when attempting to re-create orders notification channel

hichamboushaba opened the pull request:
### Description
When working on the Cha-ching fix in #11095, I added a logic to re-create the notification channel if it was still using the old Uri format (with resource ID), but the logic is causing an NPE crash if the sound was disabled instead. 

### Testing instructions
There is nothing to test, just code review to confirm the logic is correct.


- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Fix NPE when attempting to re-create orders notification channel

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/199">17.8 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/NotificationChannelsHandler.kt:
```diff
@@ -82,7 +82,7 @@ class NotificationChannelsHandler @Inject constructor(
         val sound = channel.sound
         var updatedChannel = channel
-        if (sound.toString().matches("^.*\\d+$".toRegex())) {
+        if (sound?.toString()?.matches("^.*\\d+$".toRegex()) == true) {
```
I'm checking againt `true` here because we don't want to re-create the channel if the sound was disabled, as we don't want to override the user's choice.

0nko reviewed (approved):
:shipit:

hichamboushaba committed:
Update check before re-creating order notification channel

This adds check to make sure the old notification channel was still using the default sound.

hichamboushaba commented:
@0nko sorry for this, but can you take another look, I pushed one additional commit 7395450eec3679f0945f7a97b2e3028e843d53f4, after looking a second time at our check, I realized we might override the notification sound if the user modified it by a custom sound that ends with digits, so I updated the regex to check that the current sound still uses the scheme `android.resource`.

hichamboushaba commented:
@samiuelson, please take note of this PR that we merged to the release branch. But there is no need for a new beta, since we'll have the final release tomorrow, and it should be enough.

samiuelson commented:
> @samiuelson, please take note of this PR that we merged to the release branch. But there is no need for a new beta, since we'll have the final release tomorrow, and it should be enough.

I'm forwarding this to @backwardstruck who will be finalizing the 17.8 release.
